drivers/xen/grant-table.c

```diff
@@ -1241,11 +1241,18 @@ int gnttab_map_refs(struct gnttab_map_grant_ref *map_ops,
 		case GNTST_okay:
 		{
 			struct xen_page_foreign *foreign;
+			int page_cnt;
 
 			SetPageForeign(pages[i]);
 			foreign = xen_page_foreign(pages[i]);
 			foreign->domid = map_ops[i].dom;
 			foreign->gref = map_ops[i].ref;
+			page_cnt = page_count(pages[i]);
+			if (page_cnt > FOREIGN_MAX_PAGE_COUNT) {
+				pr_warn("page have too many users. Will wait for 0 on umap\n");
```

Comment:
`WARN_ONCE` maybe?

Also, we need some comments on why this will work.

Reply:
refactored it to use printk_ratelimited.